docs/index.rst

```diff
@@ -6,8 +6,220 @@
+- checks (holds the :code:`check_` functions)
+
+Both of them need to be *accessible to the zoo*.
+
+Metadata
+--------
+
+Metadata is a yaml file with a list of errors your checks yield. There are two compulsory parameters in the header:
+
+- :code:`namespace`
+- :code:`category`
+
+Then you list the errors with parameters:
+
+- :code:`id` - a unique identifier (*required*)
+- :code:`title` - a human-readable title (*required*)
+- :code:`description` - additional information about the error (*optional*)
+
+**Metadata Structure:**
+
+.. literalinclude:: ..\zoo\auditing\standards\dummy_standards\metadata\check_constantly.yml
+    :language: yaml
+
+**Real life example:**
+
+.. code-block:: yaml
+
+    namespace: coafile
+    category: Configuration of coala
+    ---
+    - id: create_coafile
+      severity: warning
+      effort: high
+      title: Use coala to run static analysis checks on your repo
+      description: |
+        Add a `.coafile` so that people can see what rules and checks they should follow
+        when contributing to your repository,
+        and run [coala](https://coala.io) to find and fix these issues automatically.
+
+    - id: dont_use_default_section
+      severity: advice
+      effort: low
+      title: Your coafile uses the deprecated [Default] section
+      description: |
+        As of [coala 0.11](https://github.com/coala/coala/releases/tag/0.11.0),
+        usage of a `[Default]` section is deprecated,
+        since the config file now supports proper inheritance definitions.
+
+        To apply options to all sections, rename `[Default]` to `[all]`,
+        and add `all.` before every other section's name,
+        to make them inherit `[all]`'s settings.
+
+Checks
+------
+
+The zoo starts all functions in this file that have the prefix :code:`check_` (if you need any helper function just name them anything else).
+
+All :code:`check_` functions need to have an input parameter :code:`context`.
+
+:code:`context` is an object of CheckContext class (`definition <https://github.com/kiwicom/the-zoo/blob/master/zoo/auditing/runner.py>`_), the object consists of all the required information about the code that is checked.
+
+The most important parameter of the class is :code:`path`, it holds the information about where the source code is exposed.
+
+All :code:`check_` functions need to return an iterator. The recommended way is to yield generator as soon as you discover errors.
+
+The structure of the generator is: :code:`context.Result("namespace:id", True/False)` where
+
+- :code:`namespace` is the namespace from your metadata file
+- :code:`id` is the id of the raised error from your metadata file
+- :code:`True/False` determines if the error was found (true if it was, false if it was not).
+
+Check function structure:
+
+.. literalinclude:: ..\zoo\auditing\standards\dummy_standards\checks\check_constantly.py
+    :language: python
+
+Real life example:
+
+.. code-block:: python
```

Comment:
I think this example could be simplified. It's too much stuff that it's not directly related to the checks and people might be confused. I know I provided you with the example but I didn't realize that issue. I'll provide you a new one.

Reply:
Can you please send me the new example 🙏

Reply:
Yep, sorry. I forgot do give you another example. I'd just show something really simple like:

```python
def check_dockerignore(context):
    if not (context.path / "Dockerfile").is_file():
        return

    try:
        dockerignore_text = (context.path / ".dockerignore").read_text()
    except FileNotFoundError:
        dockerignore_text = ""  # no dockerignore is effectively an empty dockerignore

    yield context.Result("dockerignore:create_dockerignore", not dockerignore_text)
    yield context.Result("dockerignore:ignore_dotgit", ".git" not in dockerignore_text)
```